infrastructure/main.tf

```diff
@@ -56,6 +56,13 @@ resource "azurerm_key_vault_secret" "POSTGRES_DATABASE" {
   key_vault_id = data.azurerm_key_vault.key_vault.id
 }
 
+resource "azurerm_management_lock" "resource-group-level" {
+  name       = "resource-group-level"
+  scope      = local.shared_infra_rg.id
+  lock_level = "CanNotDelete"
+  notes      = "This Resource Group can't be deleted"
+}
+
```

Comment:
Instead of creating a resource lock in here. Should we create RG lock in the database module itself? In that ways we will not have to update each and every repo. Whoever is using the database module will have the lock automatically enabled for the database resource group.

Reply:
Yes as discussed we would be maintaining the locks via a dedicated pipeline which will cover the resource groups containing blob storage, Key Vault, Storage account and DB.